File: lib/common/dto/location/location_datasource_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:line_flux/common/model/location/location_data_source.dart';

part 'location_datasource_dto.freezed.dart';
part 'location_datasource_dto.g.dart';

@freezed
class LocationDataSourceDTO with _$LocationDataSourceDTO {
  const LocationDataSourceDTO._();

  factory LocationDataSourceDTO.fromJson(Map<String, dynamic> json) =>
      _$LocationDataSourceDTOFromJson(json);

  const factory LocationDataSourceDTO({
    required String id,
  }) = _LocationDataSourceDTO;


  factory LocationDataSourceDTO.fromDataModel(LocationDataSource datasource) {
    return LocationDataSourceDTO(id: datasource.id);
  }

  LocationDataSource toDataModel() {
    return LocationDataSource();
  }
}

File: android/app/src/main/kotlin/com/line_flux/line_flux/dto/data_source/LocationDataSourceDTO.kt
package com.line_flux.line_flux.dto.data_source

import com.kotlin.sensor_drawing_plugin.datasource.LocationDataSource

data class LocationDataSourceDTO(val id: String) {
  companion object {
    fun fromDataModel(dataSource: LocationDataSource): LocationDataSourceDTO {
      return LocationDataSourceDTO(id = dataSource.id)
    }
  }

  fun toDataModel(): LocationDataSource {
    val dataSource = LocationDataSource()
    dataSource.id = id
    return dataSource
  }
}
